extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/datatransferproject/transfer/smugmug/photos/SmugMugPhotosImporter.java

```diff
@@ -141,4 +154,68 @@ private SmugMugInterface getOrCreateSmugMugInterface(TokenSecretAuthData authDat
         ? new SmugMugInterface(appCredentials, authData, mapper)
         : smugMugInterface;
   }
+
+  /**
+   * Key for cache of album mappings. TODO: Add a method parameter for a {@code key} for fine
+   * grained objects.
+   */
+  private String createCacheKey() {
+    // TODO: store objects containing individual mappings instead of single object containing all
+    // mappings
+    return "tempPhotosData";
+  }
+
+  /**
+   * Get the proper album upload information for the photo. Takes into account size limits of the
+   * albums and completed uploads.
+   */
+  private SmugMugPhotoTempData getAlbumTempData(
+      UUID jobId, IdempotentImportExecutor idempotentExecutor, String baseAlbumId)
+      throws Exception {
+    SmugMugPhotoTempData baseAlbumTempData =
+        jobStore.findData(jobId, baseAlbumId, SmugMugPhotoTempData.class);
+    SmugMugPhotoTempData albumTempData = baseAlbumTempData;
+    int depth = 0;
+    while (albumTempData.getPhotoCount() >= transmogrificationConfig.getAlbumMaxSize()) {
+      if (albumTempData.getOverflowAlbumId() == null) {
+        PhotoAlbum newAlbum =
+            createOverflowAlbum(
+                baseAlbumTempData.getAlbumId(),
+                baseAlbumTempData.getAlbumName(),
+                baseAlbumTempData.getAlbumDescription(),
+                depth + 1);
+        String newUri =
+            idempotentExecutor.executeOrThrowException(
+                newAlbum.getId(),
+                newAlbum.getName(),
+                () -> importSingleAlbum(jobId, newAlbum));
+        albumTempData.setOverflowAlbumId(newAlbum.getId());
+        jobStore.update(jobId, albumTempData.getAlbumId(), albumTempData);
+        albumTempData =
+            jobStore.findData(
+                jobId, albumTempData.getOverflowAlbumId(), SmugMugPhotoTempData.class);
+      } else {
+        albumTempData =
+            jobStore.findData(
+                jobId, albumTempData.getOverflowAlbumId(), SmugMugPhotoTempData.class);
+      }
+      depth += 1;
+    }
+    return albumTempData;
+  }
+
+  /**
+   * Create an overflow album id -> {baseAlbumId}-overflow-{copyNumber} name -> {baseAlbumName}
+   * ({copyNumber}) description -> {baseAlbumDescription}
+   */
```

Comment:
Could you make this comment a little clearer? Maybe just state what the expected album is going to be?

Reply:
yeah